### PR TITLE
Cleanup properties & vasc/properties

### PR DESCRIPTION
**Pull request by mgeplf**
* use macro to reduce templated code
* functions exist in headers, and should be inlined in each TU
* SomaLevel was never used
* CellLevel wasn't being completely constructed

### CODE REVIEW

File: include/morphio/vasc/properties.h [hunk 1]
```diff
@@ -68,7 +69,7 @@ struct Properties {
     std::vector<Connection::Type> _connectivity;
 
     template <typename T>
-    std::vector<typename T::Type>& get() noexcept;
+    std::vector<typename T::Type>& get_mut() noexcept;
```
Comment (tomdele) [inline include/morphio/vasc/properties.h:72]: wouldn't be clearer to have : 
```
get
cget (or get_const)
```
as names instead ? I feel like the `mut` suffix collides a bit with the "mutable morphologies" naming convention. 

But this is maybe personal and a question of taste.

File: include/morphio/properties.h [hunk 2]
```diff
@@ -206,59 +194,42 @@ struct Properties {
...
-template <>
-const std::vector<MitoSection::Type>& Properties::get<MitoSection>() const noexcept;
-template <>
-std::vector<MitoSection::Type>& Properties::get<MitoSection>() noexcept;
-
-template <>
-const std::vector<MitoPathLength::Type>& Properties::get<MitoPathLength>() const noexcept;
-template <>
-std::vector<MitoPathLength::Type>& Properties::get<MitoPathLength>() noexcept;
+#define INSTANTIATE_TEMPLATE_GET(T, M)                                       \
```
Comment (tomdele) [inline include/morphio/properties.h:200]: are the `\` mandatory ?

**Review by tomdele — COMMENTED** [review]
Just a question/discussion about naming otherwise it looks good.

File: include/morphio/properties.h (hunk 2 above)
Comment (matz-e) [reply include/morphio/properties.h:200]: yes

File: include/morphio/properties.h [hunk 3]
```diff
@@ -206,59 +194,42 @@ struct Properties {
...
-template <>
-std::vector<MitoDiameter::Type>& Properties::get<MitoDiameter>() noexcept;
+#undef INSTANTIATE_TEMPLATE_GET
 
 template <>
-const std::vector<Section::Type>& Properties::get<Section>() const noexcept;
-template <>
-std::vector<Section::Type>& Properties::get<Section>() noexcept;
+inline const std::map<int32_t, std::vector<uint32_t>>& Properties::children<Section>() const
+    noexcept {
```
Comment (matz-e) [inline include/morphio/properties.h:224]: What kind of line wrapping is this? Readability suffers a bit, I was wondering at first what a `noexcept` block is…

File: include/morphio/vasc/properties.h (hunk 1 above)
Comment (matz-e) [reply include/morphio/vasc/properties.h:72]: `const` by default seems like a better interface…

**Review by matz-e — APPROVED** [review]
Seems like a nice cleanup. Now if only that macro stuff could be deduplicated…

File: include/morphio/properties.h (hunk 3 above)
Comment (mgeplf) [reply include/morphio/properties.h:224]: clang-format

**Pull request merged**